x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/HttpClientManager.java

```diff
@@ -128,29 +128,34 @@ public HttpClient getHttpClient() {
     @Override
     public void close() throws IOException {
         httpClient.close();
-        connectionEvictor.stop();
+        connectionEvictor.close();
     }
 
     private void setMaxConnections(int maxConnections) {
         connectionManager.setMaxTotal(maxConnections);
         connectionManager.setDefaultMaxPerRoute(maxConnections);
     }
 
+    // This is only used for testing
+    boolean isEvictionThreadRunning() {
+        return connectionEvictor.isRunning();
+    }
+
     // default for testing
     void setEvictionInterval(TimeValue evictionInterval) {
+        logger.error("called set eviction interval " + evictionInterval);
         evictorSettings = new EvictorSettings(evictionInterval, evictorSettings.evictionMaxIdle);
 
-        connectionEvictor.stop();
+        connectionEvictor.close();
         connectionEvictor = createConnectionEvictor();
         connectionEvictor.start();
     }
 
     void setEvictionMaxIdle(TimeValue evictionMaxIdle) {
-        evictorSettings = new EvictorSettings(evictorSettings.evictionInterval, evictionMaxIdle);
+        logger.error("called set eviction max idle " + evictionMaxIdle);
 
-        connectionEvictor.stop();
-        connectionEvictor = createConnectionEvictor();
-        connectionEvictor.start();
+        evictorSettings = new EvictorSettings(evictorSettings.evictionInterval, evictionMaxIdle);
+        connectionEvictor.setMaxIdleTime(evictionMaxIdle);
```

Comment:
We don't need to restart the evictor when this setting changes since it's just used when the evictor runs.